### ISSUE TITLE
Struct literals

### ISSUE THREAD

**Issue by segeljakt** [cda-group/arc#28]
Structs (#27) are defined in Arc using the abstract syntax:

```
type ::= struct-type
       | ...

struct-type ::= '{' type (',' type)* '}'

expr ::= struct-literal
       | struct-index
       | ...

struct-literal ::= '{' expr (',' expr)* '}'
struct-index ::= expr '.$'[0-9]+
```

Some observations:
* `struct-type` maps directly to `tuple<type+>` in MLIR.
* `struct-literal` and `struct-index` are not offered by MLIR ([rationale](https://github.com/tensorflow/mlir/blob/master/g3doc/Rationale.md#tuple-types)), so we need to implement them ourselves. I propose that we start with literals.

**Comment by segeljakt** [comment]
I think I managed to get this working O.o

Will open PR after #23 is merged

**Comment by segeljakt** [comment]
Addressed in #38 and #47.
